src/dlls/dbgshim/dbgshim.cpp

```diff
@@ -1052,6 +1053,56 @@ IsCoreClrWithGoodHeader(
     return false;
 }
 
+static
+HMODULE*
+EnumProcessModulesInternal(
+    HANDLE hProcess, 
+    DWORD *pCountModules)
+{
+    *pCountModules = 0;
+
+    // Start with 1024 modules
+    DWORD cbNeeded = sizeof(HMODULE) * 1024;
+
+    ArrayHolder<BYTE> modules = new (nothrow) BYTE[cbNeeded];
+    if (modules == nullptr)
+    {
+        SetLastError(ERROR_OUTOFMEMORY);
+        return nullptr;
+    }
+
+    if(!EnumProcessModules(hProcess, (HMODULE*)modules.GetPtr(), cbNeeded, &cbNeeded))
+    {
+        return nullptr;
+    }
+
+    // If 1024 isn't enough, try the modules array size returned (cbNeeded)
+    if (cbNeeded > (sizeof(HMODULE) * 1024)) 
+    {
+        modules = new (nothrow) BYTE[cbNeeded];
+        if (modules == nullptr)
+        {
+            SetLastError(ERROR_OUTOFMEMORY);
+            return nullptr;
+        }
+
+        DWORD cbNeeded2;
+        if(!EnumProcessModules(hProcess, (HMODULE*)modules.GetPtr(), cbNeeded, &cbNeeded2))
+        {
+            return nullptr;
+        }
+
+        // The only way cbNeeded2 could change on the second call is if more
+        // modules were loaded by the process in the small window between the
+        // above EnumProcessModules calls. If this actually happens, then give
+        // up on trying to get the whole module list and risk missing the coreclr
+        // module.
```

Comment:
Don't you need: `cbNeeded = min(cbNeeded, cbNeeded2);`

Reply:
Yes, I just realized that.  I assumed that it could only get larger and didn't that it could get smaller.